examples/apps/echo/client/echo_client.py

```diff
@@ -257,6 +269,20 @@ def Main(args=None):
 				decrypted_res = utility.decrypted_response(
 					res, session_key, session_iv)
 				logger.info("\nDecrypted response:\n {}".format(decrypted_res))
+				if input_data_hash:
+					# Verify data hash in response
+					decrypted_data = decrypted_res[0]["data"]
+					computed_data_hash = utility.compute_data_hash(decrypted_data)
+					# Convert data_hash to hex
+					computed_data_hash = ''.join(format(i, '02x') \
+						for i in computed_data_hash).upper()
+					data_hash_in_resp = (decrypted_res[0]["dataHash"]).upper()
+					if computed_data_hash == data_hash_in_resp:
+						logger.info("Computed hash of decrypted data matched "\
+							"with data hash in the response")
+					else:
+						logger.error("Computed hash of decrypted data does not"\
+							"match with data hash in the response")
```

Comment:
Would you like to move data hash verification logic to a function for better readability ? May be check the return value of the function and exit if verification fails.

Reply:
Could be done, it doesn't much reduce the lines (lines 275 and 277 can be avoided.).  We still need to get decrypted data and hash value from the response and still need to have validation in place.

Reply:
The function can return a boolean. You can also move the logging inside this function which is fine as this will be an internal function.

Reply:
Done